gnssanalysis/gn_io/sp3.py

```diff
@@ -85,104 +78,86 @@ def mapparm(old, new):
     return off, scl
 
 
+def _process_sp3_block(date, data, widths, names):
+    """Process a single block of SP3 data"""
+    if not data or len(data) == 0:
+        return _pd.DataFrame()
+    epochs_dt = _pd.to_datetime(_pd.Series(date).str.slice(2, 21).values.astype(str), format=r"%Y %m %d %H %M %S")
+    temp_sp3 = _pd.read_fwf(_io.StringIO(data), widths=widths, names=names)
+    dt_index = _np.repeat(a=_gn_datetime.datetime2j2000(epochs_dt.values), repeats=len(temp_sp3))
+    temp_sp3.set_index(dt_index, inplace=True)
+    temp_sp3.index.name = "J2000"
+    temp_sp3.set_index(temp_sp3.PRN.astype(str), append=True, inplace=True)
+    temp_sp3.set_index(temp_sp3.PV_FLAG.astype(str), append=True, inplace=True)
+    return temp_sp3
+
+
 def read_sp3(sp3_path, pOnly=True, nodata_to_nan=True):
```

Comment:
The output type-hint is missing here. 
Also, I think @EugeneDu-GA should test any sp3 files he was having trouble with against this function